src/tribler/gui/core_manager.py

```diff
@@ -196,18 +202,24 @@ def kill_core_process_and_remove_the_lock_file(self):
         process_checker = ProcessChecker(self.root_state_dir)
         process_checker.remove_lock()
 
+    def get_last_core_output(self, quoted=True):
+        output = ''.join(self.last_core_stderr_output) or ''.join(self.last_core_stdout_output)
+        if quoted:
+            output = re.sub(r'^', '> ', output, flags=re.MULTILINE)
+        return output
+
     @staticmethod
-    def format_error_message(exit_code: int, exit_status: int, last_core_output: str) -> str:
+    def format_error_message(exit_code: int, exit_status: int) -> str:
         message = f"The Tribler core has unexpectedly finished with exit code {exit_code} and status: {exit_status}."
-        try:
-            string_error = os.strerror(exit_code)
-        except ValueError:
-            # On platforms where strerror() returns NULL when given an unknown error number, ValueError is raised.
-            string_error = 'unknown error number'
+        if exit_code == 1:
+            string_error = "Application error"
+        else:
+            try:
+                string_error = os.strerror(exit_code)
+            except ValueError:
+                # On platforms where strerror() returns NULL when given an unknown error number, ValueError is raised.
+                string_error = 'unknown error number'
         message += f'\n\nError message: {string_error}'
-
-        quoted_output = re.sub(r'^', '> ', last_core_output, flags=re.MULTILINE)
-        message += f"\n\nLast core output:\n{quoted_output}"
```

Comment:
This PR moves the rendering of the last core output to `gui_error`, as we need it not only for `CoreCrashedError` but for other subclasses of CoreError as well, for example, for `CoreConnectTimeoutError`.